sim/warlock/rotations.go

```diff
@@ -159,28 +228,44 @@ func (warlock *Warlock) defineRotation() {
 	}
 }
 
+/*
+	At the end of this function, every spell in our arsenal is given;
+	 A castIn() value, that determines when the spell will be needed to/will be ready to recast again.
+	 A Priority value, that lets us order which one of the "Ready Spells(aka. castIn = 0)" we will be using.
+*/
+
+// Regen Spells: Casts the regen spell that will give you the most mana, includes a error whenever we cast pact on full mana.
 func (warlock *Warlock) LifeTapOrDarkPact(sim *core.Simulation) {
 	if warlock.CurrentManaPercent() == 1 {
 		panic("Life Tap or Dark Pact while full mana")
 	}
-	if warlock.Talents.DarkPact && warlock.Pet.CurrentMana() > warlock.GetStat(stats.SpellPower)+1200+131 {
+	if warlock.Talents.DarkPact && warlock.Pet.CurrentMana() > warlock.GetStat(stats.SpellPower)+1200+131 { //Evaluates based on your SP, if DP or LT will give you the highest mana.
 		warlock.DarkPact.Cast(sim, warlock.CurrentTarget)
 	} else {
 		warlock.LifeTap.Cast(sim, warlock.CurrentTarget)
 	}
 }
 
+// This function is an intermediary, it is used when sim has a GCD ready, not much to see here.
 func (warlock *Warlock) OnGCDReady(sim *core.Simulation) {
 	warlock.tryUseGCD(sim)
 }
 
+//preparation function definitions ends::
+
+/*
+This function is the way we execute the main functionality of this entire script.
+All of the previously implemented functions come together in this function.
+Function takes the Warlock character that's used to model the client behavior, and returns the "modified" simulation state.
+Might sound complicated, worry not, things will get better.
+*/
 func (warlock *Warlock) tryUseGCD(sim *core.Simulation) {
-	var spell *core.Spell
-	var filler *core.Spell
-	var target = warlock.CurrentTarget
-	mainSpell := warlock.Rotation.PrimarySpell
-	curse := warlock.Rotation.Curse
-	dotLag := time.Duration(10 * time.Millisecond)
+	var spell *core.Spell                          //the variable we'll be returning to the sim as our final decision
+	var filler *core.Spell                         //the filler spell we'll store, we will cast this whenever we have all our priorities in check
+	var target = warlock.CurrentTarget             //our current target
+	mainSpell := warlock.Rotation.PrimarySpell     // our primary spell
```

Comment:
Well I did not change its name from the TBC sim as SB & Incinerate were primary spells, but they are basically filler spells now rather than primary spells.

Reply:
I believe "filler" and "main" spells being used interchangeably is not that big of a problem because both are correct definitions.  An afflictions main source of damage is SB for example, yet it's not a priority. All the while you cast it at the bottom of the cast hierarchy.